src/MagnumPlugins/KtxImageConverter/KtxImageConverter.cpp

```diff
@@ -0,0 +1,932 @@
+    offset += dfdSamplesSize;
+
+    Utility::copy(sampleData.second(), samples);
+
+    UnsignedShort extent = 0;
+    for(auto& sample: samples) {
+        extent = Math::max<UnsignedShort>(sample.bitOffset + sample.bitLength + 1, extent);
+        Utility::Endianness::littleEndianInPlace(sample.bitOffset,
+            sample.lower, sample.upper);
+    }
+
+    /* Just making sure we didn't make any major mistake in samples() */
+    CORRADE_INTERNAL_ASSERT(extent == header.bytesPlane[0]*8);
+
+    Utility::Endianness::littleEndianInPlace(length);
+    Utility::Endianness::littleEndianInPlace(header.vendorId, header.descriptorType,
+        header.versionNumber, header.descriptorBlockSize);
+
+    CORRADE_INTERNAL_ASSERT(offset == dfdSize);
+
+    return data;
+}
+
+template<std::size_t Size> struct TypeForSize {};
+template<> struct TypeForSize<1> { typedef UnsignedByte  Type; };
+template<> struct TypeForSize<2> { typedef UnsignedShort Type; };
+template<> struct TypeForSize<4> { typedef UnsignedInt   Type; };
+template<> struct TypeForSize<8> { typedef UnsignedLong  Type; };
+
+void endianSwap(Containers::ArrayView<char> data, UnsignedInt typeSize) {
+    switch(typeSize) {
+        case 1:
+            /* Single-byte or block-compressed format, nothing to do */
+            return;
+        case 2:
+            Utility::Endianness::littleEndianInPlace(Containers::arrayCast<TypeForSize<2>::Type>(data));
+            return;
+        case 4:
+            Utility::Endianness::littleEndianInPlace(Containers::arrayCast<TypeForSize<4>::Type>(data));
+            return;
+        case 8:
+            Utility::Endianness::littleEndianInPlace(Containers::arrayCast<TypeForSize<8>::Type>(data));
+            return;
+    }
+
+    CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+}
+
+UnsignedInt leastCommonMultiple(UnsignedInt a, UnsignedInt b) {
+    const UnsignedInt product = a*b;
+
+    /* Greatest common divisor */
+    while(b != 0) {
+        const UnsignedInt t = a % b;
+        a = b;
+        b = t;
+    }
+    const UnsignedInt gcd = a;
+
+    return product/gcd;
+}
+
+template<UnsignedInt dimensions>
+void copyPixels(const BasicImageView<dimensions>& image, Containers::ArrayView<char> pixels) {
+    std::size_t sizes[dimensions + 1];
+    sizes[dimensions] = image.pixelSize();
+    for(UnsignedInt i = 0; i != dimensions; ++i) {
+        sizes[dimensions - 1 - i] = image.size()[i];
+    }
+
+    /* Copy the pixels into output, dropping padding (if any) */
+    Utility::copy(image.pixels(), Containers::StridedArrayView<dimensions + 1, char>{pixels, sizes});
+}
+
+template<UnsignedInt dimensions>
+void copyPixels(const BasicCompressedImageView<dimensions>& image, Containers::ArrayView<char> pixels) {
+    /** @todo How do we deal with CompressedPixelStorage::skip?
+              ImageView::pixels handles this for non-compressed images. */
```

Comment:
Is it enough to assert that `image.storage() == CompressedPixelStorage{}`? Because the block (data) size in there is also ignored and a mismatch should be an error.

Reply:
Yes, that should be enough :)